src/anemoi/inference/clusters/cluster.py

```diff
@@ -0,0 +1,183 @@
+# (C) Copyright 2025- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+#
+
+import datetime
+import logging
+import os
+from abc import ABC
+from abc import abstractmethod
+from typing import Any
+from typing import NamedTuple
+
+import numpy as np
+
+from anemoi.inference.context import Context
+from anemoi.inference.lazy import torch
+
+ADDRESS = NamedTuple("Address", [("host", str), ("port", int)])
+LOG = logging.getLogger(__name__)
+
+
+class Cluster(ABC):
+    """Abstract base class for cluster and parallel environment handling."""
+
+    _model_comm_group: torch.distributed.ProcessGroup | None = None  # type: ignore
```

Comment:
one thought here: I think torch.distributed will be eventually superseded by monarch. But then that may take year+